sycl/include/sycl/buffer.hpp

```diff
@@ -438,6 +438,8 @@ class buffer : public detail::buffer_plain,
           PI_ERROR_INVALID_VALUE);
   }
 
+  buffer &operator=(buffer<T, dimensions, AllocatorT> &rhs) = delete;
```

Comment:
```suggestion
  buffer &operator=(const buffer &rhs) = delete;
```
Also, why do we delete it? We have a valid copy ctor for this class below.

Reply:
Line 461 of this file has that operator= definition already; Coverity complains about a missing user-defined assignment operator for the templated definition of buffer as shown in the diff.

Reply:
It might not be necessary to have this templated assignment operator defined, but Coverity checked all of the copy constructor parameters and reported this as a problem.  This addition will be removed.